### PR TITLE
Tmedia 15 float left right article body 

### PR DESCRIPTION
**Pull request by JackHowa**
## Description
On Desktop, float images left or right based on position 

## Jira Ticket
- [TMEDIA-15](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-15&assignee=5e387d6a1b1d910e5dfd7a9b)

## Acceptance Criteria
When photos within the article body has a right or left float applied, the photo should render on the right or left side of desktop/tablet view, like the example screenshot shows above. 

Text should wrap around the photo with appropriate 24px margin spacing

Center and none floats remain as-is (full width)

Mobile view remains as-is (no float or text wrapping on mobile)

Limiting height for portrait orientation photos will be addressed in TMEDIA-321: Limit height of photo embeds within article bodyPO GROOMING and should be released together

Tests and documentation are updated as necessary

## Test Steps
1. Checkout this branch `git checkout TMEDIA-15-float-left-right-article-body`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/article-body-block`
3. Go to and see text move around the old-fashion image  http://localhost/pf/entertainment/2021/06/28/jenaes-global-kitchen-sink-article/?_website=the-gazette

### Before

Screen Shot 2021-07-16 at 14 29 22

### After

Screen Shot 2021-07-16 at 14 27 51

## Deferred
- This is only on desktop

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/article-body-block/chains/article-body/default.jsx [hunk 1]
```diff
@@ -32,6 +32,36 @@ const StyledLink = styled.a`
   color: ${(props) => props.primaryColor};
 `;
 
+// IMPROVEMENT: margin and margin bottom rem is reusable
+const FloatableImageContainer = styled.figure`
+  margin: 0 0 0 1rem;
```
Comment (matthewroach) [inline blocks/article-body-block/chains/article-body/default.jsx]: This is applying a margin left to all images in the article body and causing non floated images to not align as the currently do with the article text.

Microsoft Edge-2021-07-19-11-06 55

File: blocks/article-body-block/chains/article-body/default.jsx [hunk 2]
```diff
@@ -32,6 +32,36 @@ const StyledLink = styled.a`
...
+
+  @media screen and (min-width: 48rem) {
+    margin-bottom: 1.5rem;
+  }
+
+  ${({ allowedFloatValue }) => allowedFloatValue && css`
```
Comment (matthewroach) [inline blocks/article-body-block/chains/article-body/default.jsx]: The ticket mentions this should not be applied to "mobile" breakpoint. This is floating the image at all breakpoints

File: blocks/article-body-block/chains/article-body/default.jsx [hunk 3]
```diff
@@ -77,19 +107,44 @@ function parseArticleItem(item, index, arcSite, phrases, id, customFields) {
...
+        widthsObject = {
+          small: 274,
+          medium: 400,
+          large: 768,
+        };
+      }
+
       return (url && url.length > 0) ? (
-        <figure key={key}>
+        <FloatableImageContainer
```
Comment (matthewroach) [inline blocks/article-body-block/chains/article-body/default.jsx]: Curious to the reason for using styled components over classnames here?

**Review by matthewroach — CHANGES_REQUESTED** [review]
A few small items: 

* Float right is working as expected.
* Float left and none has additional left margin
* Applied to all breakpoints?

File: blocks/article-body-block/chains/article-body/default.jsx (hunk 1 above)
Comment (JackHowa) [reply blocks/article-body-block/chains/article-body/default.jsx]: ah this is a typo

Comment (JackHowa) [reply blocks/article-body-block/chains/article-body/default.jsx]: thanks, this was a syntax mistake should be only bottom

Comment (JackHowa) [reply blocks/article-body-block/chains/article-body/default.jsx]: updated

File: blocks/article-body-block/chains/article-body/default.jsx (hunk 2 above)
Comment (JackHowa) [reply blocks/article-body-block/chains/article-body/default.jsx]: updated

Comment (JackHowa) [reply blocks/article-body-block/chains/article-body/default.jsx]: my bad, think I didn't push the latest commits 🤦

File: blocks/article-body-block/chains/article-body/default.jsx (hunk 3 above)
Comment (JackHowa) [reply blocks/article-body-block/chains/article-body/default.jsx]: I can refactor it to use sass; I thought at least the margin and bottom changes could be reusable

File: blocks/article-body-block/chains/article-body/default.jsx [hunk 4]
```diff
@@ -32,6 +32,34 @@ const StyledLink = styled.a`
   color: ${(props) => props.primaryColor};
 `;
 
+// IMPROVEMENT: margin and margin bottom rem is reusable
+const FloatableImageContainer = styled.figure`
+  margin: 0 0 1rem 0;
```
[the remark this answers is not in the document]
Comment (JackHowa) [reply blocks/article-body-block/chains/article-body/default.jsx]: updated

File: blocks/article-body-block/chains/article-body/_articlebody.scss [hunk 5]
```diff
@@ -281,3 +279,19 @@
...
+  }
+}
+
+.article-body-image-container--right-float {
+  @media screen and (min-width: 48rem) {
+    width: 50%;
+    float: right;
+    margin-left: 1.5rem;
+  }
+}
```
[the remark this answers is not in the document]
Comment (JackHowa) [reply blocks/article-body-block/chains/article-body/_articlebody.scss]: resolved

**Comment by JackHowa** [comment]
Updated @matthewroach 


"desktop": 

with float 

desktop-left-float

with no float

desktop no float

"mobile" 

with float alignment, does not float on mobile breakpoint

alignment-left-mobile

with no float

Screen Shot 2021-07-19 at 10 47 46

**Review by matthewroach — APPROVED** [review]
Nice! Thanks for the updates 🎉

**Pull request merged**